### PR TITLE
modified oper.g for use by the Utils package

### PR DESCRIPTION
**Pull request by ChrisJefferson**
This is an updated version of #408 by @cdwensley .

Changes:

1) Add test, improve documentation (remove implementation details), make sure documentation is linked into the manual, make sure example passes 'make testmanuals'.

### CODE REVIEW

**Comment by fingolfin** [comment]
I am confused by this. The PR #408 by @cdwensley you point to has this description:
```
When moving global functions from other packages to Utils it is desirable for the code declaring and implementing these functions to be duplicated in the two packages. The additions to oper.g are designed to ensure that DeclareGlobalFunction and InstallGlobalFunction are called only once for functions being transferred. Detailed discussion of this procedure is given in the Introduction and Appendix to the manual in version 0.13 of Utils (available very soon). The intention is that these additions will be removed before GAP 4.8 is made publicly available. 
```

So, it already announces that those changes should be backed out before a public GAP release.

But then, for whom are those changes? End users would never ever see them, right? So only package authors would be affected. 

But why do these people need those changes in the first place? I do understand that there is an awkward period of time where stuff is moved from package XYZ to Utils, but no new versions of Utils and XYZ are released. The changes to GAP seem to be meant to make that transition period a bit less awward for package authors.

But if the author of XYZ fully cooperates with the Utils package team anyway, can't they simply resolve this by following these steps:
* add functions to Utils
* make a new Utils release, version N.M
* once that is "out" (i.e. available on @cdwensley 's homepage), change package XYZ to require Utils N.M or newer
* make a new XYZ release
* and only then tell the GAP package distribution software about the new versions of Utils and XYZ, so that end users never see an termporary incompatible state.

Short version: I don't understand whom this PR is supposed to help, and how.

**Comment by cdwensley** [comment]
The comment "The intention is that these additions will be removed before GAP 4.8 is made publicly available." was made some time ago, when I naively hoped that PR408 would be included in 4.8.1 and that all the transfers would take place before 4.8.2.  It now seems likely that PR539 will not be committed until at least 4.8.3, with transfers to Utils following that. 
As to the steps listed above, it is probably more convenient for package authors if the Utils-N.M is publicly released before the new version of XYZ is produced - it certainly gives them more time. 
It has also been suggested that the new functions in the PR might be useful for other transfers, unrelated to Utils, and so would remain for as long as they are considered useful.

**Comment by frankluebeck** [comment]
This looks all very complicated. 

Why cannot the Util package include code from package Bla, version <= x.y  by putting

```
if TestPackageAvailability("Bla", ">=x.(y+1)") = fail then
   ...
fi;
```
around that code. Later, when users have Bla in version at least x.(y+1) this can be removed.

**Comment by cdwensley** [comment]
The idea behind this latest suggestion will be investigated for Utils. 
TestPackageAvailability is not the correct function to use, since it returns true/false if the package is loaded; the InstallationPath if not yet loaded; and fail if the package is not available. 
It is important to be able to test the code from Bla in Utils without having Bla already loaded. 
The function InstalledPackageVersion would appear to provide the required information.

**Comment by frankluebeck** [comment]
@cdwensley:  the documentation of `TestPackageAvailability` says

_So the package name is available if the result of TestPackageAvailability is not equal to fail._

Therefore, I think that the code snippet I gave above is correct.

**Comment by cdwensley** [comment]
Experimental version 0.21 of Utils has been released today for Stefan and Sebastian to test, based on Frank's suggestion rather than the modified global.g, oper.g.

**Comment by fingolfin** [comment]
@cdwensley So, can this PR and also PR #408 be closed then?

**Comment by cdwensley** [comment]
Although utils-0.21 has been produced using Frank's approach, without the changes considered here, I have had no feedback from Stefan or Sebastian so far, so it could be precipitate to throw the changes away.
Furthermore, back in Dec.15, Chris Jefferson commented "4.8 is fairly close now :) and this seems like the kind of thing that we might need over time, as more things are moved between packages, so I think the plan should be it will live in GAP forever." 
No doubt, if these PRs are closed, they could be recovered in the future if the need arose.

**Comment by alex-konovalov** [comment]
@fingolfin Maybe we need a label "on hold" for such situations?

**Comment by fingolfin** [comment]
I don't think so -- I'll just close these items, and @cdwensley can either reopen them, or open new ones, if the need arises.
